snuba/web/split.py

```diff
@@ -290,22 +285,13 @@ def execute(
             if match:
                 return None
 
-        total_legacy_cols = query.get_all_referenced_columns()
-        total_col_count = len(total_legacy_cols)
         # We need to count the number of table/column name pairs
         # not the number of distinct Column objects in the query
         # so to avoid counting aliased columns multiple times.
-        total_ast_cols = {
+        total_columns = {
             (col.table_name, col.column_name)
             for col in query.get_all_ast_referenced_columns()
         }
-        total_ast_count = len(total_ast_cols)
-        if total_col_count != total_ast_count:
-            logger.warning(
```

Comment:
This was a limitation of the legacy query. It was counting literals as columns. So the right result is the one from the AST